### DISCUSSION TITLE
[ADDON-CONTROLS / ADDON-KNOBS] Generate "options" dynamically of an argType

### DISCUSSION THREAD

**Discussion by X976647** [storybookjs/storybook#23789] [Help]
Hi,
I have a problem with addon-controls. I want to configure the args and argsType using just this addon, but I have a property whose values depend on the theme that is selected and I need to pick the options dynamically.

I could make it with addon-knobs, but I read that is going to be deprecated. My story is something like this right now:

//With knobs
```
const Template: StoryFn<typeof ContainerStory> = (args: IContainer) => {
  const theme = useTheme();
  const variants = theme[VARIANT_TYPE];
  const variantOptions= [...Object.values(variants)];
  const variant = select("variant", variantOptions, variantOptions[0]);

  return (
    <ContainerStory
      {...args}
      variant={variant}
    >
      CHILDREN
    </ContainerStory>
  );
};
```

I would like to get something using argTypes with controls and complete the options in the story dynamically.

I'd appreciate your help.
Thanks!

**Comment by omarkawach** [comment]
Also interested in this. Sad to see addon-knobs deprecated.
